### DISCUSSION TITLE
Help with 2d geometry basics

### DISCUSSION THREAD

**Discussion by Dizzixx** [CadQuery/cadquery#1330] [Q&A]
I think I am probably missing something simple and yet fundamental I am trying to assemble what is in effect a chain of lines. Some of which are simple and others of which might be an arc, spline, etc. So there is a list of points and a connectivity table defining the segments and their type. 

I can make a polyline easy enough which treats the connection between all points as straight lines easy enough. And I can make a series of independent lines also. But I want them to nest inside of a parent object like they do for the polyline. 

I have tried chaining things together using the output from each prior call as the object calling pointTo() at the next but that did not do what I though it would.

Below is a minimum working example illustrating what I have tried and if launched in the CQ-editor should illustrate what I mean about multiple line objects versus one polyline object. Attached is the test csv file used for the points and connectivity.

Any help would be greatly appreciated. 

```
with open("test.csv","r") as f:
    lines = [line for line in f]

nPoints = int(lines[0].strip().split(',')[0])

xs,ys=[],[]
for i in range(nPoints):
    x,y = (float(e) for e in lines[i+1].strip().split(',')[:2])
    xs.append(x*.5)
    ys.append(-y)
    # remember to halve diameters
    # and flip the y values to invert

cons = []
for i in range(nPoints+1,len(lines)):
    lines[i].strip().split(',')
    cons.append(
        [float(e) for e in 
         lines[i].strip().split(',')
         if not e==""])

if cons[-1][1]!=0.:
    cons.append([len(cons),0.,1,1])

print(cons)


import cadquery as cq

poly = (
        cq.Workplane("XY")
        .polyline(
            [(x,y) for x,y in zip(
                [*xs,xs[0]], # closed
                [*ys,ys[0]]) # poly
                ])
        )
show_object(poly)

cqLines = []
result = None
Workplane = cq.Workplane("XY")

for c_id,con in enumerate(cons):
    print(c_id,con)

    
    i,j,t = con[:3]
    x0,y0 = xs[int(i)],ys[int(i)]
    x1,y1 = xs[int(j)],ys[int(j)]
    dx,dy = x1-x0,y1-y0

    #if lineSet is None: 
    #   lineSet=cq.Workplane("XY")
    if c_id==0:
        result = cq.Workplane("XY")
        print(result)
    if con[3]!=3:
        cqLines.append(
            Workplane.center(x0,y0)
            .lineTo(dx,dy)
            )
        result = result.center(x0,y0).lineTo(dx,dy)
    else:
        cqLines.append(
            Workplane.center(x0,y0)
            .radiusArc((dx,dy),-con[4])
            # have to flip sign of r
            # depending on concavity
            )
        result=result.center(x0,y0).radiusArc((dx,dy),-con[4])
    print("As result",result)
    print("As lines:",cqLines[-1])
    
    
#show_object(Workplane)
# Render the lines
show_object(result)
for line in cqLines:show_object(line)
```

[test.csv](https://github.com/CadQuery/cadquery/files/11378068/test.csv)

**Comment by jmwright** [comment]
You might want to look at the Sketch API, specifically the edge-based part of it.

https://cadquery.readthedocs.io/en/latest/sketch.html#edge-based-api
